### PR TITLE
Project controller RSpec (part 5)

### PR DESCRIPTION
**Pull request by mdeniz**
Fix name creation for repository factory

### CODE REVIEW

File: src/api/spec/controllers/webui/project_controller_spec.rb [hunk 1]
```diff
@@ -403,4 +403,57 @@
...
+      expect(assigns(:linking_projects)).to match_array([another_project.name])
+    end
+  end
+
+  describe 'DELETE #destroy' do
+    before do
+      login user_moi
+    end
+
+    context 'with check_weak_dependencies to true' do
```
Comment (bgeuken) [inline src/api/spec/controllers/webui/project_controller_spec.rb]: *with check_weak_dependencies enabled

File: src/api/spec/controllers/webui/project_controller_spec.rb [hunk 2]
```diff
@@ -403,4 +403,57 @@
...
+          delete :destroy, project: user_moi.home_project
+        end
+
+        it { expect(Project.count).to eq(0) }
+        it { is_expected.to redirect_to(action: :index) }
+        it { expect(flash[:notice]).to eq("Project was successfully removed.") }
+      end
+    end
+
+    context 'with check_weak_dependencies to false' do
```
Comment (bgeuken) [inline src/api/spec/controllers/webui/project_controller_spec.rb]: and disabled

File: src/api/spec/controllers/webui/project_controller_spec.rb [hunk 3]
```diff
@@ -456,4 +457,41 @@
       it { expect(flash[:notice]).to start_with("Project can't be removed:") }
     end
   end
+
+  describe 'POST #update_target' do
+    before do
+      login user_moi
```
Comment (bgeuken) [inline src/api/spec/controllers/webui/project_controller_spec.rb]: AFAIK this needs to be "login user" once rebased on master

File: src/api/spec/controllers/webui/project_controller_spec.rb [hunk 4]
```diff
@@ -416,7 +416,7 @@
 
   describe 'DELETE #destroy' do
     before do
-      login user_moi
+      login user
```
Comment (bgeuken) [inline src/api/spec/controllers/webui/project_controller_spec.rb]: Nevermind my previous comment. Though would you mind to squash this into the previous commit(s)?

**Comment by bgeuken** [comment]
LGTM :+1:

File: src/api/spec/controllers/webui/project_controller_spec.rb (hunk 4 above)
Comment (mdeniz) [inline src/api/spec/controllers/webui/project_controller_spec.rb]: Ok I will squash them

**Pull request merged**